ldes-server-domain/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/domain/converter/FragmentIdConverter.java

```diff
@@ -1,15 +1,20 @@
 package be.vlaanderen.informatievlaanderen.ldes.server.domain.converter;
 
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.ldesfragmentrequest.entities.FragmentPair;
+
+import java.util.List;
+
 public class FragmentIdConverter {
     private FragmentIdConverter() {
     }
 
-    public static String toFragmentId(String hostLocation, String view, String fragmentationType,
-                                      String fragmentationValue) {
-        return "%s/%s?%s=%s".formatted(hostLocation, view, getCompactTimestampPath(fragmentationType), fragmentationValue);
-    }
-
-    private static String getCompactTimestampPath(String timestampPath) {
-        return timestampPath.split("#")[1];
+    public static String toFragmentId(String hostLocation, String view, List<FragmentPair> fragmentPairs) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("%s/%s".formatted(hostLocation, view));
+        if(!fragmentPairs.isEmpty()){
+            stringBuilder.append("?");
+            fragmentPairs.forEach(fragmentPair -> stringBuilder.append("%s=%s".formatted(fragmentPair.fragmentKey(), fragmentPair.fragmentValue())));
```

Comment:
Just as a general point, do we plan to limit what can be the fragmentvalue? I am picturing a string value that contains '"' or '=' which could break the url

Reply:
Hmm, that's correct. Will add an issue so that we won't forget to tackle this in a follow up PR.